webui/react/src/hooks/useModal/useModalExperimentGroups.tsx

```diff
@@ -0,0 +1,115 @@
+import { DeleteOutlined } from '@ant-design/icons';
+import { Button, Form, Input } from 'antd';
+import { ModalFuncProps } from 'antd/es/modal/Modal';
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
+
+import { deleteExperimentGroup, getExperimentGroups, patchExperimentGroup } from 'services/api';
+import useModal, { ModalHooks } from 'shared/hooks/useModal/useModal';
+import { alphaNumericSorter } from 'shared/utils/sort';
+import { ExperimentGroup } from 'types';
+import handleError from 'utils/error';
+
+import css from './useModalExperimentGroups.module.scss';
+
+interface Props {
+  onAction?: () => void;
+  onClose?: () => void;
+  projectId: number;
+}
+
+const useModalExperimentGroups = ({ onAction, onClose, projectId }: Props): ModalHooks => {
+  const { modalOpen: openOrUpdate, modalRef, ...modalHook } = useModal({ onClose });
+  const [form] = Form.useForm();
+  const [experimentGroups, setExperimentGroups] = useState<ExperimentGroup[]>([]);
+  const [canceler] = useState(new AbortController());
+
+  const fetchExperimentGroups = useCallback(async () => {
+    try {
+      const groups = await getExperimentGroups({ projectId }, { signal: canceler.signal });
+      groups.sort((a, b) => alphaNumericSorter(a.name, b.name));
+      setExperimentGroups(groups);
+    } catch (e) {
+      handleError(e);
+    }
+  }, [canceler.signal, projectId]);
+
+  const handleRenameClick = useCallback(
+    async (groupId: number) => {
+      try {
+        const name = (form.getFieldValue([groupId, 'name']) as string).trim();
+        await patchExperimentGroup({ groupId, name, projectId }, { signal: canceler.signal });
+        await fetchExperimentGroups();
+        onAction?.();
+      } catch (e) {
+        handleError(e);
+      }
+    },
+    [canceler.signal, fetchExperimentGroups, form, onAction, projectId],
+  );
+
+  const handleDeleteClick = useCallback(
+    async (groupId: number) => {
+      try {
+        await deleteExperimentGroup({ groupId, projectId }, { signal: canceler.signal });
+        await fetchExperimentGroups();
+        onAction?.();
+      } catch (e) {
+        handleError(e);
+      }
+    },
+    [canceler.signal, fetchExperimentGroups, onAction, projectId],
+  );
+
+  const modalContent = useMemo(() => {
+    return (
+      <Form className={css.container} form={form} layout="vertical">
+        {experimentGroups.map((group) => (
```

Comment:
When opening up the Groups modal and there are 0 groups in this project, can we add a message that there are no Groups?